ScarletEditor/Sources/EditorMain.cpp

```diff
@@ -7,44 +7,36 @@
 #include "RenderModule.h"
 #include "UIModule.h"
 
+#include "AchillesDemo.h"
+
+using namespace ScarletEngine;
+
 int main()
 {
-	using namespace ScarletEngine;
-	
 	// #todo_Core: this should be loaded by a config file or something, for now default it to this.
 	AssetManager::SetAssetRoot("../");
 
 	ModuleManager::GetInstance().RegisterModule<RenderModule>();
 	ModuleManager::GetInstance().RegisterModule<UIModule>();
 
+	ScarletEngine::Achilles::RegisterSystems();
+
 	GEngine = MakeUnique<Engine>();
 	GEngine->Initialize();
 
 	// initialize the global editor
 	GEditor = MakeUnique<Editor>();
 	// #todo_core: this should be handled automatically by the engine
 	GEditor->Initialize();
-
 	{
-		// Test entity
-		auto [Ent, Trans, Mesh] = GEditor->GetActiveWorld()->CreateEntity<Transform, StaticMeshComponent>("Monkey");
-
-		Trans->Position = glm::vec3(0.f, 0.5f, 0.f);
-		Trans->Rotation = glm::vec3(0.f, 0.f, 0.f);
-		Trans->Scale = glm::vec3(0.5f);
-
-		Mesh->MeshHandle = AssetManager::LoadStaticMesh("/ScarletEngine/Content/Monkey.obj");
-		Mesh->VertexBuff = RAL::Get().CreateBuffer((uint32_t)Mesh->MeshHandle->Vertices.size() * sizeof(Vertex), RALBufferUsage::STATIC_DRAW);
-		Mesh->VertexBuff->UploadData(Mesh->MeshHandle->Vertices.data(), Mesh->MeshHandle->Vertices.size() * sizeof(Vertex));
-		Mesh->IndexBuff = RAL::Get().CreateBuffer((uint32_t)Mesh->MeshHandle->Indices.size() * sizeof(uint32_t), RALBufferUsage::STATIC_DRAW);
-		Mesh->IndexBuff->UploadData(Mesh->MeshHandle->Indices.data(), Mesh->MeshHandle->Indices.size() * sizeof(uint32_t));
-		Mesh->VertexArray = RAL::Get().CreateVertexArray(Mesh->VertexBuff, Mesh->IndexBuff);
-
-		auto VertShader = RAL::Get().CreateShader(RALShaderStage::Vertex, "/ScarletEngine/Shaders/test_shader.vert");
-		auto FragShader = RAL::Get().CreateShader(RALShaderStage::Pixel, "/ScarletEngine/Shaders/test_shader.frag");
-		Mesh->Shader = RAL::Get().CreateShaderProgram(VertShader, FragShader, nullptr, nullptr);
-		ScarDelete(VertShader);
-		ScarDelete(FragShader);
+		//ScarletEngine::Achilles::DemoKepler(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoSpherePlaneCollision(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoSphereSphereCollision(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoSpring(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoRope(GEditor->GetActiveWorld(), glm::vec3(0.25f, 0.f, 0.f));
+		ScarletEngine::Achilles::DemoRopeCollision(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoFlag(GEditor->GetActiveWorld());
+		//ScarletEngine::Achilles::DemoSoftBodies(GEditor->GetActiveWorld());
```

Comment:
We don't need all these commented out lines here, it just makes it harder to read.